### PR TITLE
Squashing port-to-postgresql further

### PR DESCRIPTION
**Pull request by brucebolt**
This branch was created from `port-to-postgresql-squashed` (PR #1196,  a partially-cleaned version of #1085 ) yesterday, and squashed down to 39 commits by @brucebolt and @JonathanHallam. However, the commit messages still showed lots of artefacts from previous messy rebases - duplicated messages, or messages that no longer accurately described the changes.

I (@aldavidson) went through all the commits this today, editing & rewording messages to create a cleaner history, and I also found a few more commits that could be squashed together. It's now ready for re-review.

This application is owned by the publishing platform team. Please let us know in #govuk-publishing-platform when you raise any PRs.   

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/models/content_item.rb
Comment (brucebolt) [inline app/models/content_item.rb:5]: I think the changes in this commit can be squashed into the previous one.

Maybe rubocop needs running after each commit.  I may give that a go and see what happens, as it'll make all the commits easier to review.

**Review by brucebolt — COMMENTED** [review]
This all looks great to me!  A heroic effort migrating to PostgreSQL.

I've run rubocop after each commit, to reduce the number of diffs in later commits and rebased against main.

I can't approve the PR as I opened it.  So my suggestion would be to push the `test` branch to a better named branch, then open another PR (titled "Port application to PostgreSQL", or similar), which I can approve.

**Review by richardTowers — APPROVED** [review]
I had a read through this commit-by-commit (thanks for cleaning them up), and it all looks good to me.

I had one extremely minor nitpick comment about a comment, but other than that I'm happy for this to merge. Approving so you can choosed to merge this PR if you like, but if you want to open another PR with a better branch name that's fine too - let me know and I'll approve that too.

**Comment by aldavidson** [comment]
Note for the record: we've decided the best way to resolve the situation  (production being built from `port-to-postgresql`, but this branch `test` being the cleaner and rebased one) is as follows:

1. force-push `test` to origin/`port-to-postgresql` (PR #1085)
2. merge #1085 into `main`

This will allow us to maintain all the existing links & references to #1085 in Github comments / commit messages / documents on Google drive / Trello cards, etc as the primary source of truth and history.

**Comment by aldavidson** [comment]
I've force-pushed this branch to port-to-postgresql (#1085) - so I'll close this unmerged. All tests have passed on #1085, so as soon as there's an approval on that, I can merge it

**Pull request closed without merging**
